core/mavlink_system.cpp

```diff
@@ -553,7 +553,10 @@ void MAVLinkSystem::get_param_float_async(const std::string &name,
                                             callback));
 }
 
-MAVLinkCommands::Result MAVLinkSystem::set_flight_mode(FlightMode system_mode)
+MAVLinkCommands::Result
+MAVLinkSystem::make_command_flight_mode(FlightMode flight_mode,
+                                        uint8_t component_id,
+                                        MAVLinkCommands::CommandLong &command)
```

Comment:
What is the reason to only make the command but not send it yet?

Reply:
I like the separation of concerns: one function makes it, the other one sends it. For instance, `set_flight_mode(...)` reads "make the command, and if it results in a success, then send it". When reading `set_flight_mode(...)`, I might not be interested at all in the detail of `make_command_flight_mode`. And I can go there if I am.

Reply:
@JonasVautherin You worded my thought. :) 
Yeah, `MAVLinkCommands` class provides `send_command(..)` so, I thought lets directly call that with the command prepared from `make_command_*` stuff.